From fb00a84b3dce58b80ac758cba2427067660aa03d Mon Sep 17 00:00:00 2001
From: Elvis Lee <kwangwoong.lee@lge.com>
Date: Wed, 26 Apr 2017 12:29:24 +0900
Subject: [PATCH] Avoid loading comments from JPEG and PNG files

:Release Notes:
Skip reading JPEG and PNG comments information from the header to save
the memory.

:Detailed Notes:
JPEG or PNG file may have very large comments information in its header
and there exist actual cases in webOS TV. (Some splash images require
4MB+ each for comments information.)
Assuming that there is no use case of this information in webOS, we'd
better to skip reading the comments information and save the memory.
Details are described in the issue addressed.

:Testing Performed:
GLD 1321, 1339
1. long run launch test by script.
2. manual launch test to check images are shown as before.

:QA Notes:

:Issues Addressed:
[PLAT-38674] LSM memory increased after some tests compare to DRD4TV

Upstream-Status : Inappropriate [webOS specific]

Change-Id: I3744312f69aa3201d5188776cbd99fe690b75d32
---
 src/gui/image/qjpeghandler.cpp | 2 ++
 src/gui/image/qpnghandler.cpp  | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/src/gui/image/qjpeghandler.cpp b/src/gui/image/qjpeghandler.cpp
index 68709b7..09554d3 100644
--- a/src/gui/image/qjpeghandler.cpp
+++ b/src/gui/image/qjpeghandler.cpp
@@ -908,6 +908,7 @@ bool QJpegHandlerPrivate::readJpegHeader(QIODevice *device)
 
             for (jpeg_saved_marker_ptr marker = info.marker_list; marker != NULL; marker = marker->next) {
                 if (marker->marker == JPEG_COM) {
+#ifdef WEBOS_SUPPORT_IMAGE_TEXT
                     QString key, value;
                     QString s = QString::fromLatin1((const char *)marker->data, marker->data_length);
                     int index = s.indexOf(QLatin1String(": "));
@@ -923,6 +924,7 @@ bool QJpegHandlerPrivate::readJpegHeader(QIODevice *device)
                     description += key + QLatin1String(": ") + value.simplified();
                     readTexts.append(key);
                     readTexts.append(value);
+#endif
                 } else if (marker->marker == JPEG_APP0 + 1) {
                     exifData.append((const char*)marker->data, marker->data_length);
                 }
diff --git a/src/gui/image/qpnghandler.cpp b/src/gui/image/qpnghandler.cpp
index e9944e1..0e00257 100644
--- a/src/gui/image/qpnghandler.cpp
+++ b/src/gui/image/qpnghandler.cpp
@@ -491,6 +491,7 @@ static void CALLBACK_CALL_TYPE qt_png_warning(png_structp /*png_ptr*/, png_const
 
 void Q_INTERNAL_WIN_NO_THROW QPngHandlerPrivate::readPngTexts(png_info *info)
 {
+#ifdef WEBOS_SUPPORT_IMAGE_TEXT
     png_textp text_ptr;
     int num_text=0;
     png_get_text(png_ptr, info, &text_ptr, &num_text);
@@ -513,6 +514,7 @@ void Q_INTERNAL_WIN_NO_THROW QPngHandlerPrivate::readPngTexts(png_info *info)
         readTexts.append(value);
         text_ptr++;
     }
+#endif
 }
 
 
